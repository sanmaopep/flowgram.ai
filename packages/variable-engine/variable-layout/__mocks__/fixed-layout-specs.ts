import { FlowDocumentJSON } from '@flowgram.ai/document';

export const fixLayout1: FlowDocumentJSON = {
  nodes: [
    {
      type: 'start',
      meta: {
        isStart: true,
      },
      id: 'start',
    },
    {
      type: 'getRecords',
      id: 'getRecords_07e97c55832',
    },
    {
      type: 'loop',
      id: 'forEach_260a8f85ff2',
      blocks: [
        {
          type: 'createRecord',
          id: 'createRecord_8f85ff2c11d',
        },
        {
          type: 'dynamicSplit',
          id: 'exclusiveSplit_ff2c11d0fb4',
          blocks: [
            {
              id: 'branch_f2c11d0fb42',
            },
            {
              id: 'branch_2c11d0fb42c',
            },
          ],
        },
      ],
    },
    {
      type: 'dynamicSplit',
      id: 'exclusiveSplit_88dbf2c60ae',
      meta: {
        defaultExpanded: true,
      },
      blocks: [
        {
          id: 'branch_8dbf2c60aee',
          blocks: [
            {
              type: 'dynamicSplit',
              id: 'exclusiveSplit_a59afaadc9a',
              blocks: [
                {
                  id: 'branch_59afaadc9ac',
                },
                {
                  id: 'branch_9afaadc9acd',
                  blocks: [
                    {
                      type: 'deleteRecords',
                      id: 'deleteRecords_c32807e97c5',
                    },
                  ]
                },
              ],
            },
          ]
        },
        {
          id: 'branch_dbf2c60aee4',
          blocks: [
            {
              type: 'updateRecords',
              id: 'updateRecords_7ed2a172c32',
            },
          ]
        },
      ],
    },
    {
      type: 'end',
      id: 'end',
    },
  ],
};
