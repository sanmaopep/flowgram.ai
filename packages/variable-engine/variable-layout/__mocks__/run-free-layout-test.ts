import { describe, expect, test } from 'vitest';
import { VariableEngine } from '@flowgram.ai/variable-core';
import { ASTKind } from '@flowgram.ai/variable-core';
import {
  FlowNodeVariableData
} from '../src';
import { TestConfig, getContainer } from './container';
import { WorkflowDocument, WorkflowJSON } from '@flowgram.ai/free-layout-core';

export const runFreeLayoutTest = (testName: string, spec: WorkflowJSON, config?: TestConfig) => {
  describe(testName, async () => {
    const container = getContainer('free', config);
    const flowDocument = container.get(WorkflowDocument);
    const variableEngine = container.get(VariableEngine);

    variableEngine.onScopeChange(action => {
      const { type, scope } = action;

      // 作用域默认创建一个默认变量
      if (type === 'add') {
        scope.ast.set('/', {
          kind: ASTKind.VariableDeclaration,
          type: ASTKind.String,
          key: String(scope.id),
        });
      }
    });

    // 创建一个全局作用域
    variableEngine.createScope('testScope');

    const traverseVariableDatas = () =>
      flowDocument
        .getAllNodes()
        // TODO 包含 $ 的节点不注册 variableData
        .filter(_node => !_node.id.startsWith('$'))
        .map(_node => _node.getData(FlowNodeVariableData))
        .filter(Boolean);

    // 初始化所有节点的私有作用域
    const initAllNodePrivate = () => {
      traverseVariableDatas().forEach(_data => {
        _data.initPrivate();
      });
    };

    // 初始化所有节点的可用变量
    const printAllNodeAvailableMapping = (_scopeType: 'public' | 'private' = 'public') =>
      traverseVariableDatas().reduce((acm, _data) => {
        const scope = _data[_scopeType]!;
        acm.set(String(scope.id), scope.available.variableKeys);

        return acm;
      }, new Map<string, string[]>());

    // 初始化所有节点的覆盖作用域
    const printAllCovers = (_scopeType: 'public' | 'private' = 'public') =>
      traverseVariableDatas().reduce((acm, _data) => {
        const scope = _data[_scopeType]!;
        acm.set(
          String(scope.id),
          scope.coverScopes.map(_scope => String(_scope.id)),
        );

        return acm;
      }, new Map<string, string[]>());

    await flowDocument.fromJSON(spec);

    test('test get Deps', () => {
      expect(printAllNodeAvailableMapping()).toMatchSnapshot();
    });

    test('test get Covers', () => {
      expect(printAllCovers()).toMatchSnapshot();
    });

    test('test get Deps After Init Private', () => {
      initAllNodePrivate();
      expect(printAllNodeAvailableMapping()).toMatchSnapshot();
    });

    test('test get private scope Deps', () => {
      expect(printAllNodeAvailableMapping('private')).toMatchSnapshot();
    });

    test('test get Covers After Init Private', () => {
      expect(printAllCovers()).toMatchSnapshot();
    });

    test('test get private scope Covers', () => {
      expect(printAllCovers('private')).toMatchSnapshot();
    });

    test('test sort', () => {
      expect(variableEngine.getAllScopes({ sort: true }).map(_scope => _scope.id)).toMatchSnapshot();
    });

    config?.runExtraTest?.(container);
  });


}
