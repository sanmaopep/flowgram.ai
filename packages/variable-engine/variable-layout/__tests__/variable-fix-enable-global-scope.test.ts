import { runFixedLayoutTest } from '../__mocks__/run-fixed-layout-test';
import { fixLayout1 } from '../__mocks__/fixed-layout-specs';

runFixedLayoutTest('Variable Fix Layout Enable Global Scope', fixLayout1, {
  enableGlobalScope: true,
});
