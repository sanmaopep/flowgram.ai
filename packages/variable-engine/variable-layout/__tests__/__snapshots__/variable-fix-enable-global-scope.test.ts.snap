// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Variable Fix Layout Enable Global Scope > test get Covers 1`] = `
Map {
  "start" => [
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "createRecord_8f85ff2c11d",
    "exclusiveSplit_ff2c11d0fb4",
    "branch_f2c11d0fb42",
    "branch_2c11d0fb42c",
    "exclusiveSplit_88dbf2c60ae",
    "branch_8dbf2c60aee",
    "exclusiveSplit_a59afaadc9a",
    "branch_59afaadc9ac",
    "branch_9afaadc9acd",
    "deleteRecords_c32807e97c5",
    "branch_dbf2c60aee4",
    "updateRecords_7ed2a172c32",
    "end",
  ],
  "getRecords_07e97c55832" => [
    "forEach_260a8f85ff2",
    "createRecord_8f85ff2c11d",
    "exclusiveSplit_ff2c11d0fb4",
    "branch_f2c11d0fb42",
    "branch_2c11d0fb42c",
    "exclusiveSplit_88dbf2c60ae",
    "branch_8dbf2c60aee",
    "exclusiveSplit_a59afaadc9a",
    "branch_59afaadc9ac",
    "branch_9afaadc9acd",
    "deleteRecords_c32807e97c5",
    "branch_dbf2c60aee4",
    "updateRecords_7ed2a172c32",
    "end",
  ],
  "forEach_260a8f85ff2" => [
    "exclusiveSplit_88dbf2c60ae",
    "branch_8dbf2c60aee",
    "exclusiveSplit_a59afaadc9a",
    "branch_59afaadc9ac",
    "branch_9afaadc9acd",
    "deleteRecords_c32807e97c5",
    "branch_dbf2c60aee4",
    "updateRecords_7ed2a172c32",
    "end",
  ],
  "createRecord_8f85ff2c11d" => [
    "exclusiveSplit_ff2c11d0fb4",
    "branch_f2c11d0fb42",
    "branch_2c11d0fb42c",
  ],
  "exclusiveSplit_ff2c11d0fb4" => [],
  "branch_f2c11d0fb42" => [
    "branch_2c11d0fb42c",
  ],
  "branch_2c11d0fb42c" => [],
  "exclusiveSplit_88dbf2c60ae" => [
    "end",
  ],
  "branch_8dbf2c60aee" => [
    "branch_dbf2c60aee4",
    "updateRecords_7ed2a172c32",
  ],
  "exclusiveSplit_a59afaadc9a" => [],
  "branch_59afaadc9ac" => [
    "branch_9afaadc9acd",
    "deleteRecords_c32807e97c5",
  ],
  "branch_9afaadc9acd" => [],
  "deleteRecords_c32807e97c5" => [],
  "branch_dbf2c60aee4" => [],
  "updateRecords_7ed2a172c32" => [],
  "end" => [],
}
`;

exports[`Variable Fix Layout Enable Global Scope > test get Covers After Init Private 1`] = `
Map {
  "start" => [
    "getRecords_07e97c55832",
    "getRecords_07e97c55832_private",
    "forEach_260a8f85ff2",
    "forEach_260a8f85ff2_private",
    "createRecord_8f85ff2c11d",
    "createRecord_8f85ff2c11d_private",
    "exclusiveSplit_ff2c11d0fb4",
    "exclusiveSplit_ff2c11d0fb4_private",
    "branch_f2c11d0fb42",
    "branch_f2c11d0fb42_private",
    "branch_2c11d0fb42c",
    "branch_2c11d0fb42c_private",
    "exclusiveSplit_88dbf2c60ae",
    "exclusiveSplit_88dbf2c60ae_private",
    "branch_8dbf2c60aee",
    "branch_8dbf2c60aee_private",
    "exclusiveSplit_a59afaadc9a",
    "exclusiveSplit_a59afaadc9a_private",
    "branch_59afaadc9ac",
    "branch_59afaadc9ac_private",
    "branch_9afaadc9acd",
    "branch_9afaadc9acd_private",
    "deleteRecords_c32807e97c5",
    "deleteRecords_c32807e97c5_private",
    "branch_dbf2c60aee4",
    "branch_dbf2c60aee4_private",
    "updateRecords_7ed2a172c32",
    "updateRecords_7ed2a172c32_private",
    "end",
    "end_private",
  ],
  "getRecords_07e97c55832" => [
    "forEach_260a8f85ff2",
    "forEach_260a8f85ff2_private",
    "createRecord_8f85ff2c11d",
    "createRecord_8f85ff2c11d_private",
    "exclusiveSplit_ff2c11d0fb4",
    "exclusiveSplit_ff2c11d0fb4_private",
    "branch_f2c11d0fb42",
    "branch_f2c11d0fb42_private",
    "branch_2c11d0fb42c",
    "branch_2c11d0fb42c_private",
    "exclusiveSplit_88dbf2c60ae",
    "exclusiveSplit_88dbf2c60ae_private",
    "branch_8dbf2c60aee",
    "branch_8dbf2c60aee_private",
    "exclusiveSplit_a59afaadc9a",
    "exclusiveSplit_a59afaadc9a_private",
    "branch_59afaadc9ac",
    "branch_59afaadc9ac_private",
    "branch_9afaadc9acd",
    "branch_9afaadc9acd_private",
    "deleteRecords_c32807e97c5",
    "deleteRecords_c32807e97c5_private",
    "branch_dbf2c60aee4",
    "branch_dbf2c60aee4_private",
    "updateRecords_7ed2a172c32",
    "updateRecords_7ed2a172c32_private",
    "end",
    "end_private",
  ],
  "forEach_260a8f85ff2" => [
    "exclusiveSplit_88dbf2c60ae",
    "exclusiveSplit_88dbf2c60ae_private",
    "branch_8dbf2c60aee",
    "branch_8dbf2c60aee_private",
    "exclusiveSplit_a59afaadc9a",
    "exclusiveSplit_a59afaadc9a_private",
    "branch_59afaadc9ac",
    "branch_59afaadc9ac_private",
    "branch_9afaadc9acd",
    "branch_9afaadc9acd_private",
    "deleteRecords_c32807e97c5",
    "deleteRecords_c32807e97c5_private",
    "branch_dbf2c60aee4",
    "branch_dbf2c60aee4_private",
    "updateRecords_7ed2a172c32",
    "updateRecords_7ed2a172c32_private",
    "end",
    "end_private",
  ],
  "createRecord_8f85ff2c11d" => [
    "exclusiveSplit_ff2c11d0fb4",
    "exclusiveSplit_ff2c11d0fb4_private",
    "branch_f2c11d0fb42",
    "branch_f2c11d0fb42_private",
    "branch_2c11d0fb42c",
    "branch_2c11d0fb42c_private",
  ],
  "exclusiveSplit_ff2c11d0fb4" => [],
  "branch_f2c11d0fb42" => [
    "branch_2c11d0fb42c",
    "branch_2c11d0fb42c_private",
  ],
  "branch_2c11d0fb42c" => [],
  "exclusiveSplit_88dbf2c60ae" => [
    "end",
    "end_private",
  ],
  "branch_8dbf2c60aee" => [
    "branch_dbf2c60aee4",
    "branch_dbf2c60aee4_private",
    "updateRecords_7ed2a172c32",
    "updateRecords_7ed2a172c32_private",
  ],
  "exclusiveSplit_a59afaadc9a" => [],
  "branch_59afaadc9ac" => [
    "branch_9afaadc9acd",
    "branch_9afaadc9acd_private",
    "deleteRecords_c32807e97c5",
    "deleteRecords_c32807e97c5_private",
  ],
  "branch_9afaadc9acd" => [],
  "deleteRecords_c32807e97c5" => [],
  "branch_dbf2c60aee4" => [],
  "updateRecords_7ed2a172c32" => [],
  "end" => [],
}
`;

exports[`Variable Fix Layout Enable Global Scope > test get Deps 1`] = `
Map {
  "start" => [
    "GlobalScope",
  ],
  "getRecords_07e97c55832" => [
    "GlobalScope",
    "start",
  ],
  "forEach_260a8f85ff2" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
  ],
  "createRecord_8f85ff2c11d" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
  ],
  "exclusiveSplit_ff2c11d0fb4" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "createRecord_8f85ff2c11d",
  ],
  "branch_f2c11d0fb42" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "createRecord_8f85ff2c11d",
    "exclusiveSplit_ff2c11d0fb4",
  ],
  "branch_2c11d0fb42c" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "createRecord_8f85ff2c11d",
    "exclusiveSplit_ff2c11d0fb4",
    "branch_f2c11d0fb42",
  ],
  "exclusiveSplit_88dbf2c60ae" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
  ],
  "branch_8dbf2c60aee" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "exclusiveSplit_88dbf2c60ae",
  ],
  "exclusiveSplit_a59afaadc9a" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "exclusiveSplit_88dbf2c60ae",
    "branch_8dbf2c60aee",
  ],
  "branch_59afaadc9ac" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "exclusiveSplit_88dbf2c60ae",
    "branch_8dbf2c60aee",
    "exclusiveSplit_a59afaadc9a",
  ],
  "branch_9afaadc9acd" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "exclusiveSplit_88dbf2c60ae",
    "branch_8dbf2c60aee",
    "exclusiveSplit_a59afaadc9a",
    "branch_59afaadc9ac",
  ],
  "deleteRecords_c32807e97c5" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "exclusiveSplit_88dbf2c60ae",
    "branch_8dbf2c60aee",
    "exclusiveSplit_a59afaadc9a",
    "branch_59afaadc9ac",
    "branch_9afaadc9acd",
  ],
  "branch_dbf2c60aee4" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "exclusiveSplit_88dbf2c60ae",
    "branch_8dbf2c60aee",
  ],
  "updateRecords_7ed2a172c32" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "exclusiveSplit_88dbf2c60ae",
    "branch_8dbf2c60aee",
    "branch_dbf2c60aee4",
  ],
  "end" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "exclusiveSplit_88dbf2c60ae",
  ],
}
`;

exports[`Variable Fix Layout Enable Global Scope > test get Deps After Init Private 1`] = `
Map {
  "start" => [
    "GlobalScope",
    "start_private",
  ],
  "getRecords_07e97c55832" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832_private",
  ],
  "forEach_260a8f85ff2" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2_private",
  ],
  "createRecord_8f85ff2c11d" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "forEach_260a8f85ff2_private",
    "createRecord_8f85ff2c11d_private",
  ],
  "exclusiveSplit_ff2c11d0fb4" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "forEach_260a8f85ff2_private",
    "createRecord_8f85ff2c11d",
    "exclusiveSplit_ff2c11d0fb4_private",
  ],
  "branch_f2c11d0fb42" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "forEach_260a8f85ff2_private",
    "createRecord_8f85ff2c11d",
    "exclusiveSplit_ff2c11d0fb4",
    "exclusiveSplit_ff2c11d0fb4_private",
    "branch_f2c11d0fb42_private",
  ],
  "branch_2c11d0fb42c" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "forEach_260a8f85ff2_private",
    "createRecord_8f85ff2c11d",
    "exclusiveSplit_ff2c11d0fb4",
    "exclusiveSplit_ff2c11d0fb4_private",
    "branch_f2c11d0fb42",
    "branch_2c11d0fb42c_private",
  ],
  "exclusiveSplit_88dbf2c60ae" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "exclusiveSplit_88dbf2c60ae_private",
  ],
  "branch_8dbf2c60aee" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "exclusiveSplit_88dbf2c60ae",
    "exclusiveSplit_88dbf2c60ae_private",
    "branch_8dbf2c60aee_private",
  ],
  "exclusiveSplit_a59afaadc9a" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "exclusiveSplit_88dbf2c60ae",
    "exclusiveSplit_88dbf2c60ae_private",
    "branch_8dbf2c60aee",
    "branch_8dbf2c60aee_private",
    "exclusiveSplit_a59afaadc9a_private",
  ],
  "branch_59afaadc9ac" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "exclusiveSplit_88dbf2c60ae",
    "exclusiveSplit_88dbf2c60ae_private",
    "branch_8dbf2c60aee",
    "branch_8dbf2c60aee_private",
    "exclusiveSplit_a59afaadc9a",
    "exclusiveSplit_a59afaadc9a_private",
    "branch_59afaadc9ac_private",
  ],
  "branch_9afaadc9acd" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "exclusiveSplit_88dbf2c60ae",
    "exclusiveSplit_88dbf2c60ae_private",
    "branch_8dbf2c60aee",
    "branch_8dbf2c60aee_private",
    "exclusiveSplit_a59afaadc9a",
    "exclusiveSplit_a59afaadc9a_private",
    "branch_59afaadc9ac",
    "branch_9afaadc9acd_private",
  ],
  "deleteRecords_c32807e97c5" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "exclusiveSplit_88dbf2c60ae",
    "exclusiveSplit_88dbf2c60ae_private",
    "branch_8dbf2c60aee",
    "branch_8dbf2c60aee_private",
    "exclusiveSplit_a59afaadc9a",
    "exclusiveSplit_a59afaadc9a_private",
    "branch_59afaadc9ac",
    "branch_9afaadc9acd",
    "branch_9afaadc9acd_private",
    "deleteRecords_c32807e97c5_private",
  ],
  "branch_dbf2c60aee4" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "exclusiveSplit_88dbf2c60ae",
    "exclusiveSplit_88dbf2c60ae_private",
    "branch_8dbf2c60aee",
    "branch_dbf2c60aee4_private",
  ],
  "updateRecords_7ed2a172c32" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "exclusiveSplit_88dbf2c60ae",
    "exclusiveSplit_88dbf2c60ae_private",
    "branch_8dbf2c60aee",
    "branch_dbf2c60aee4",
    "branch_dbf2c60aee4_private",
    "updateRecords_7ed2a172c32_private",
  ],
  "end" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "exclusiveSplit_88dbf2c60ae",
    "end_private",
  ],
}
`;

exports[`Variable Fix Layout Enable Global Scope > test get private scope Covers 1`] = `
Map {
  "start_private" => [
    "start",
  ],
  "getRecords_07e97c55832_private" => [
    "getRecords_07e97c55832",
  ],
  "forEach_260a8f85ff2_private" => [
    "forEach_260a8f85ff2",
    "forEach_260a8f85ff2_private",
    "createRecord_8f85ff2c11d",
    "createRecord_8f85ff2c11d_private",
    "exclusiveSplit_ff2c11d0fb4",
    "exclusiveSplit_ff2c11d0fb4_private",
    "branch_f2c11d0fb42",
    "branch_f2c11d0fb42_private",
    "branch_2c11d0fb42c",
    "branch_2c11d0fb42c_private",
  ],
  "createRecord_8f85ff2c11d_private" => [
    "createRecord_8f85ff2c11d",
  ],
  "exclusiveSplit_ff2c11d0fb4_private" => [
    "exclusiveSplit_ff2c11d0fb4",
    "exclusiveSplit_ff2c11d0fb4_private",
    "branch_f2c11d0fb42",
    "branch_f2c11d0fb42_private",
    "branch_2c11d0fb42c",
    "branch_2c11d0fb42c_private",
  ],
  "branch_f2c11d0fb42_private" => [
    "branch_f2c11d0fb42",
  ],
  "branch_2c11d0fb42c_private" => [
    "branch_2c11d0fb42c",
  ],
  "exclusiveSplit_88dbf2c60ae_private" => [
    "exclusiveSplit_88dbf2c60ae",
    "exclusiveSplit_88dbf2c60ae_private",
    "branch_8dbf2c60aee",
    "branch_8dbf2c60aee_private",
    "exclusiveSplit_a59afaadc9a",
    "exclusiveSplit_a59afaadc9a_private",
    "branch_59afaadc9ac",
    "branch_59afaadc9ac_private",
    "branch_9afaadc9acd",
    "branch_9afaadc9acd_private",
    "deleteRecords_c32807e97c5",
    "deleteRecords_c32807e97c5_private",
    "branch_dbf2c60aee4",
    "branch_dbf2c60aee4_private",
    "updateRecords_7ed2a172c32",
    "updateRecords_7ed2a172c32_private",
  ],
  "branch_8dbf2c60aee_private" => [
    "branch_8dbf2c60aee",
    "branch_8dbf2c60aee_private",
    "exclusiveSplit_a59afaadc9a",
    "exclusiveSplit_a59afaadc9a_private",
    "branch_59afaadc9ac",
    "branch_59afaadc9ac_private",
    "branch_9afaadc9acd",
    "branch_9afaadc9acd_private",
    "deleteRecords_c32807e97c5",
    "deleteRecords_c32807e97c5_private",
  ],
  "exclusiveSplit_a59afaadc9a_private" => [
    "exclusiveSplit_a59afaadc9a",
    "exclusiveSplit_a59afaadc9a_private",
    "branch_59afaadc9ac",
    "branch_59afaadc9ac_private",
    "branch_9afaadc9acd",
    "branch_9afaadc9acd_private",
    "deleteRecords_c32807e97c5",
    "deleteRecords_c32807e97c5_private",
  ],
  "branch_59afaadc9ac_private" => [
    "branch_59afaadc9ac",
  ],
  "branch_9afaadc9acd_private" => [
    "branch_9afaadc9acd",
    "branch_9afaadc9acd_private",
    "deleteRecords_c32807e97c5",
    "deleteRecords_c32807e97c5_private",
  ],
  "deleteRecords_c32807e97c5_private" => [
    "deleteRecords_c32807e97c5",
  ],
  "branch_dbf2c60aee4_private" => [
    "branch_dbf2c60aee4",
    "branch_dbf2c60aee4_private",
    "updateRecords_7ed2a172c32",
    "updateRecords_7ed2a172c32_private",
  ],
  "updateRecords_7ed2a172c32_private" => [
    "updateRecords_7ed2a172c32",
  ],
  "end_private" => [
    "end",
  ],
}
`;

exports[`Variable Fix Layout Enable Global Scope > test get private scope Deps 1`] = `
Map {
  "start_private" => [
    "GlobalScope",
  ],
  "getRecords_07e97c55832_private" => [
    "GlobalScope",
    "start",
  ],
  "forEach_260a8f85ff2_private" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
  ],
  "createRecord_8f85ff2c11d_private" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "forEach_260a8f85ff2_private",
  ],
  "exclusiveSplit_ff2c11d0fb4_private" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "forEach_260a8f85ff2_private",
    "createRecord_8f85ff2c11d",
  ],
  "branch_f2c11d0fb42_private" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "forEach_260a8f85ff2_private",
    "createRecord_8f85ff2c11d",
    "exclusiveSplit_ff2c11d0fb4",
    "exclusiveSplit_ff2c11d0fb4_private",
  ],
  "branch_2c11d0fb42c_private" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "forEach_260a8f85ff2_private",
    "createRecord_8f85ff2c11d",
    "exclusiveSplit_ff2c11d0fb4",
    "exclusiveSplit_ff2c11d0fb4_private",
    "branch_f2c11d0fb42",
  ],
  "exclusiveSplit_88dbf2c60ae_private" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
  ],
  "branch_8dbf2c60aee_private" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "exclusiveSplit_88dbf2c60ae",
    "exclusiveSplit_88dbf2c60ae_private",
  ],
  "exclusiveSplit_a59afaadc9a_private" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "exclusiveSplit_88dbf2c60ae",
    "exclusiveSplit_88dbf2c60ae_private",
    "branch_8dbf2c60aee",
    "branch_8dbf2c60aee_private",
  ],
  "branch_59afaadc9ac_private" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "exclusiveSplit_88dbf2c60ae",
    "exclusiveSplit_88dbf2c60ae_private",
    "branch_8dbf2c60aee",
    "branch_8dbf2c60aee_private",
    "exclusiveSplit_a59afaadc9a",
    "exclusiveSplit_a59afaadc9a_private",
  ],
  "branch_9afaadc9acd_private" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "exclusiveSplit_88dbf2c60ae",
    "exclusiveSplit_88dbf2c60ae_private",
    "branch_8dbf2c60aee",
    "branch_8dbf2c60aee_private",
    "exclusiveSplit_a59afaadc9a",
    "exclusiveSplit_a59afaadc9a_private",
    "branch_59afaadc9ac",
  ],
  "deleteRecords_c32807e97c5_private" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "exclusiveSplit_88dbf2c60ae",
    "exclusiveSplit_88dbf2c60ae_private",
    "branch_8dbf2c60aee",
    "branch_8dbf2c60aee_private",
    "exclusiveSplit_a59afaadc9a",
    "exclusiveSplit_a59afaadc9a_private",
    "branch_59afaadc9ac",
    "branch_9afaadc9acd",
    "branch_9afaadc9acd_private",
  ],
  "branch_dbf2c60aee4_private" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "exclusiveSplit_88dbf2c60ae",
    "exclusiveSplit_88dbf2c60ae_private",
    "branch_8dbf2c60aee",
  ],
  "updateRecords_7ed2a172c32_private" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "exclusiveSplit_88dbf2c60ae",
    "exclusiveSplit_88dbf2c60ae_private",
    "branch_8dbf2c60aee",
    "branch_dbf2c60aee4",
    "branch_dbf2c60aee4_private",
  ],
  "end_private" => [
    "GlobalScope",
    "start",
    "getRecords_07e97c55832",
    "forEach_260a8f85ff2",
    "exclusiveSplit_88dbf2c60ae",
  ],
}
`;

exports[`Variable Fix Layout Enable Global Scope > test sort 1`] = `
[
  Symbol(GlobalScope),
  "start_private",
  "start",
  "getRecords_07e97c55832",
  "getRecords_07e97c55832_private",
  "forEach_260a8f85ff2",
  "forEach_260a8f85ff2_private",
  "createRecord_8f85ff2c11d",
  "createRecord_8f85ff2c11d_private",
  "exclusiveSplit_ff2c11d0fb4",
  "exclusiveSplit_ff2c11d0fb4_private",
  "branch_f2c11d0fb42",
  "branch_f2c11d0fb42_private",
  "branch_2c11d0fb42c",
  "branch_2c11d0fb42c_private",
  "exclusiveSplit_88dbf2c60ae",
  "exclusiveSplit_88dbf2c60ae_private",
  "branch_8dbf2c60aee",
  "branch_8dbf2c60aee_private",
  "exclusiveSplit_a59afaadc9a",
  "exclusiveSplit_a59afaadc9a_private",
  "branch_59afaadc9ac",
  "branch_59afaadc9ac_private",
  "branch_9afaadc9acd",
  "branch_9afaadc9acd_private",
  "deleteRecords_c32807e97c5",
  "deleteRecords_c32807e97c5_private",
  "branch_dbf2c60aee4",
  "branch_dbf2c60aee4_private",
  "updateRecords_7ed2a172c32",
  "updateRecords_7ed2a172c32_private",
  "end",
  "end_private",
  "testScope",
  "$blockIcon$exclusiveSplit_ff2c11d0fb4",
  "$inlineBlocks$exclusiveSplit_ff2c11d0fb4",
  "$blockOrderIcon$branch_f2c11d0fb42",
  "$blockOrderIcon$branch_2c11d0fb42c",
  "$blockIcon$exclusiveSplit_88dbf2c60ae",
  "$inlineBlocks$exclusiveSplit_88dbf2c60ae",
  "$blockOrderIcon$branch_8dbf2c60aee",
  "$blockIcon$exclusiveSplit_a59afaadc9a",
  "$inlineBlocks$exclusiveSplit_a59afaadc9a",
  "$blockOrderIcon$branch_59afaadc9ac",
  "$blockOrderIcon$branch_9afaadc9acd",
  "$blockOrderIcon$branch_dbf2c60aee4",
]
`;
